pkg/controller/storagecluster/reconcile.go

```diff
@@ -409,3 +466,87 @@ func newCephDaemonResources(custom map[string]corev1.ResourceRequirements) map[s
 
 	return resources
 }
+
+// newStorageClassDeviceSets converts a list of StorageDeviceSets into a list of Rook StorageClassDeviceSets
+func newStorageClassDeviceSets(storageDeviceSets []ocsv1.StorageDeviceSet, topologyMap *ocsv1.NodeTopologyMap) []rook.StorageClassDeviceSet {
+	var storageClassDeviceSets []rook.StorageClassDeviceSet
+
+	for _, ds := range storageDeviceSets {
+		resources := ds.Resources
+		if resources.Requests == nil && resources.Limits == nil {
+			resources = defaults.DaemonResources["osd"]
+		}
+
+		topologyKey := ds.TopologyKey
+		topologyKeyValues := []string{}
+		noPlacement := ds.Placement.NodeAffinity == nil && ds.Placement.PodAffinity == nil && ds.Placement.PodAntiAffinity == nil
+
+		if noPlacement {
+			if topologyKey == "" {
+				topologyKey = "zone"
+			}
+			if topologyMap != nil {
+				topologyKey, topologyKeyValues = topologyMap.GetKeyValues(topologyKey)
+			}
+		}
+
+		count := ds.Count
+		replica := ds.Replica
+		if replica == 0 {
+			replica = defaults.DeviceSetReplica
+
+			// This is a temporary hack in place due to limitations
+			// in the current implementation of the OCP console.
+			// The console is hardcoded to create a StorageCluster
+			// with a Count of 3, as made sense for the previous
+			// behavior, but it cannot be updated until the next
+			// z-stream release of OCP 4.2. This workaround is to
+			// enable the new behavior while the console is waiting
+			// to be updated.
+			// TODO: Remove this behavior when OCP console is updated
+			count = count / 3
```

Comment:
I'm still somewhat concerned with the change of meaning of `count` in the API (that this workaround is undoing for the case of `replica==0`). Unless we are managing to get the change (corresponding to the removal of the hack) into the UI before our first OCS release, there's always going to be a time window where things are broken (since we can not release OCP and OCS in perfect sync).

Of course, having this hack under the condition of `replica == 0` is pretty clever in that it allows us to change the UI to use replica explicitly, and the operator would have the new logic there right away. And later we could get rid of the hack...

Just for completeness, the alternative that I discussed with @jarrpa earlier today would be leave the meaning of `count` to be the total number of disks and distribute this number across the zones (or, later, racks). It would avoid breaking API changes. But I am not 100% sure what I would consider to be the more natural API here...

Reply:
> Just for completeness, the alternative that I discussed with @jarrpa earlier today would be leave the meaning of count to be the total number of disks and distribute this number across the zones (or, later, racks). It would avoid breaking API changes. But I am not 100% sure what I would consider to be the more natural API here...

@obnoxxx Is the motivation around keeping count to mean total disks driven by maintaining API compatibility or because there's a use case where someone would need to explicitly control disk count and zones/racks count independently?